src/pages/namespace/Monitoring/Instances/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { CheckCircle2, XCircle } from "lucide-react";
+import { Table, TableBody } from "~/design/Table";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "~/design/Tooltip";
+
+import { InstanceCard } from "./instanceCard";
+import { InstanceRow } from "./Row";
+import NoResult from "./NoResult";
+import RefreshButton from "~/design/RefreshButton";
+import { ScrollArea } from "~/design/ScrollArea";
+import { useInstances } from "~/api/instances/query/get";
+import { useTranslation } from "react-i18next";
+
+export const Instances = () => {
+  const {
+    data: sucessfullInstances,
+    isFetched: isFetchedSucessfullInstances,
+    isFetching: isFetchingSucessfullinstances,
+    refetch: refetchSucessfullInstances,
+  } = useInstances({
+    limit: 10,
+    offset: 0,
+    filters: {
+      STATUS: {
+        type: "MATCH",
+        value: "complete",
+      },
+    },
+  });
+
+  const {
+    data: failedInstances,
+    isFetched: isFetchedFailedInstances,
+    isFetching: isFetchingFailedInstances,
+    refetch: refetchFailedInstances,
+  } = useInstances({
+    limit: 10,
+    offset: 0,
+    filters: {
+      STATUS: {
+        type: "MATCH",
+        value: "failed",
+      },
+    },
+  });
+
+  const { t } = useTranslation();
+  if (!isFetchedSucessfullInstances || !isFetchedFailedInstances) return null;
+  return (
+    <>
+      <InstanceCard
+        headline={t("pages.monitoring.instances.successfullExecutions.title")}
+        icon={CheckCircle2}
+        refetchButton={
+          <TooltipProvider>
+            <Tooltip>
+              <TooltipTrigger asChild>
+                <RefreshButton
+                  icon
+                  size="sm"
+                  variant="ghost"
+                  disabled={isFetchingSucessfullinstances}
+                  onClick={() => {
+                    refetchSucessfullInstances();
+                  }}
+                />
+              </TooltipTrigger>
+              <TooltipContent>
+                {t(`pages.monitoring.instances.updateTooltip`)}
+              </TooltipContent>
+            </Tooltip>
+          </TooltipProvider>
```

Comment:
Could we declare this as a `const refetchButton` or similar before the return statement and just reference it here? I think that would be more readable, because less nested.

Reply:
Sure! (89b0c95217d9070867419c1d149783d7ed36752d)